pkg/appgw/ingress_rules.go

```diff
@@ -18,31 +18,43 @@ func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, e
 	listeners := make(map[listenerIdentifier]listenerAzConfig)
 
 	// if ingress has only backend configured
-	if ingress.Spec.Backend != nil && len(ingress.Spec.Rules) == 0 {
+	if ingress.Spec.Backend != nil && ingress.Spec.TLS == nil && len(ingress.Spec.Rules) == 0 {
```

Comment:
what if WAF Policy is defined on the ingress ?

Reply:
> what if WAF Policy is defined on the ingress with backend ?

good point

Reply:
> what if WAF Policy is defined on the ingress with backend ?

This is actually covered, check the last test case in front_listener_test.go